har_dataset.py

```diff
@@ -685,34 +715,52 @@ def __prepare_motionsense__(path, split):
         [pl.col(c) + 2 * std_values[c]
             for c in set(mean_values.columns).intersection(std_values.columns)]
     )
-
+    
     print(f'Parsing {len(all_files)} files...')
     recordings = []
+    
     for i, file in enumerate(all_files):
-        subject = int(file.stem.split('_')[1])
-        class_name_full = str(file.parent.name)
-        class_name = class_name_full.split('_')[0]
+        # skip subjects according to split id list
+        if half_data==True:
+            if subject not in ids[split]:
+                continue
+            subject = int(file.stem.split('_')[1])
+            class_name_full = str(file.parent.name)
+            class_name = class_name_full.split('_')[0]
         
-        df = pl.read_csv(file)
-        df = df.drop("") # drop index col  because of duplicates
-        # add cls and subject cols
-        df = df.with_columns([
-            pl.lit(class_name_full).alias('class_name_full'),
-            pl.lit(class_name).alias('class_name'),
-            pl.lit(subject).alias('subject')
-        ])
-        # subselect split first 70% for train, next 15% for val, next 15% for test
-        total_rows = df.shape[0]
-        val_start_row = round(total_rows * 0.7)
-        test_start_row = round(total_rows * 0.85)
-
-        match split:
-            case 'train':
-                df = df[0:val_start_row]
-            case 'val':
-                df = df[val_start_row:test_start_row]
-            case 'test':
-                df = df[test_start_row:]
+            df = pl.read_csv(file)
+            df = df.drop("") # drop index col  because of duplicates
+            # add cls and subject cols
+            df = df.with_columns([
+                pl.lit(class_name_full).alias('class_name_full'),
+                pl.lit(class_name).alias('class_name'),
+                pl.lit(subject).alias('subject')
+                ])
+        else:
+            subject = int(file.stem.split('_')[1])
+            class_name_full = str(file.parent.name)
+            class_name = class_name_full.split('_')[0]
```

Comment:
Else not needed if you do the same as in mbientlab.

This should be enough
```suggestion
    subject = int(file.stem.split('_')[1])
    class_name_full = str(file.parent.name)
    class_name = class_name_full.split('_')[0]
    if half_data==True:
            if subject not in ids[split]:
                continue
```